### DISCUSSION TITLE
Как узнать версию прошивки

### DISCUSSION THREAD

**Discussion by maxpavlov** [Arkady23/V380-Web#9] [Q&A]
Добрый день. 

Купил камеры с Алиэкспресс, работают с v380 приложением.
https://www.aliexpress.com/item/1005002618555265.html?spm=a2g0s.9042311.0.0.2b104c4dJRaAL7

Протокол авторизации и дискавери такой как у уже изученных 380-ток, а вот поток получить не могу, никакой, ни RTSP ни просто в ответ, как например умеет делать с рядом камер вот этот проект:
https://github.com/prsyahmi/v380

Уже вскрыл wireshark'ом процесс общения мобильного клиента и камеры, там конечно все работает хорошо - но перед тем как перейти к последнему варианту - писать имитатор общения на основе перехваченных пакетов, хотел узнать, может таки можно как-то узнать прошивку и попробовать перепрошить камеру что бы появился RTSP. 

Любая подсказка о том как можно достучаться до камеры не через родного клиента - буду признателен.

**Comment by maxpavlov** [comment]
Получил чуть больше инфо про прошивку с SDK через перехват трафика, 
наблюдаю такую строчку
F_C_V1.0.7.420210428AC790N_SDK_V1.5.020200811A9_J320210106ÿ%2021-06-28

В самом приложении v380 вот что:
Software version: F_C_V1.0.7.4_20210428
Firmware version: A9_J3_20210106

Как думаете, имея даные выше, можно ли сделать кастомную прошику? Возможно что-то еще надо, могу снять все что угодно вайршарком. 

Заранее спасибо за помощь.

**Reply by Arkady23** [reply]
> Уже вскрыл wireshark'ом процесс общения мобильного клиента и камеры

Этот трафик не нужен. Нужен трафик камеры. Данный проект не связаан с SDK и разработкой прошивок. Это дело производителя. Предполагается, что у вас заводская последняя прошивка. На главной странице всё необходимое написано. Есть 4 способа. Самый простой снять трафик с камеры, есть видеодемонстрацмя,

**Reply by maxpavlov** [reply]
Очень благодарен вам за ответ. Данная строка, которая включает и тектс со словами SDK, так же как и другой трафик "снята с камеры" во время ее коммуникации с мобильным клиентом. Видео инструкция которую имеем обрывается еще до того как там видно получение HWXXXX, я лишь предполагаю что вы этим видео предлагаете получить полный дамп коммуникации, а далее поискать "param=". В моем сливе трафика с камеры не было текста "param" ни в одном из TCP пакетов, поэтому именно такую строчку найти не смог. Соотвественно не знаю как сделать инсталятор которым бы можно было сделать прошивку.
